system/src/ble_listening_mode_handler.cpp

```diff
@@ -124,6 +124,9 @@ int BleListeningModeHandler::restoreUserConfigurations() {
 
     // Do not allow other thread to modify the BLE configurations.
     CHECK(hal_ble_lock(nullptr));
+    SCOPE_GUARD ({
```

Comment:
Let's instead move `WiringBleLock` out of cpp file into the header and utilize RAII.